client/data/marketplace/use-wporg-plugin-query.ts

```diff
@@ -18,35 +20,41 @@ import { BASE_STALE_TIME, WPORG_CACHE_KEY } from './constants';
 import { Plugin, PluginQueryOptions } from './types';
 import { getPluginsListKey } from './utils';
 
+export const getFetchWPORGPlugins = (
+	options: PluginQueryOptions,
+	locale: string
+): [ QueryKey, QueryFunction< { plugins: Plugin[]; info: { page: number } }, QueryKey > ] => {
+	const cacheKey = getPluginsListKey( WPORG_CACHE_KEY, options );
+	const fetchFn = () => {
+		const [ search, author ] = extractSearchInformation( options.searchTerm );
+		return fetchPluginsList( {
+			pageSize: options.pageSize,
+			page: options.page,
+			category: options.category,
+			locale: options.locale || locale,
+			search,
+			author,
+			tag: options.tag && ! search ? options.tag : null,
+		} );
+	};
+	return [ cacheKey, fetchFn ];
+};
+
 export const useWPORGPlugins = (
 	options: PluginQueryOptions,
 	{ enabled = true, staleTime = BASE_STALE_TIME, refetchOnMount = true }: UseQueryOptions = {}
```

Comment:
I believe there is a better way to pass custom query options here. Instead of supporting just two (somewhat randomly chosen) options, why not allow to customize everything? A common pattern that we've been using in other Calypso queries (e.g., [`useFollowersQuery`](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/followers/use-followers-query.js)) is like:
```js
function useMyQuery( myQueryOptions, queryOptions) {
  return useQuery( key, fn, {
    ...queryOptions,
    select: ( data ) => { ... },
  } );
}
```
I.e., just spread the entire `queryOptions` object and add the non-overridable options like `select`. That should be more concise and more universal.

Reply:
Sounds good. 
I'll address this tomorrow as this seems to trigger some type mismatches for the `select` function.